Added activity functions to solid solution, updated example document

bobmyhill opened the pull request:
Quite often, we want to use the activity of a particular endmember in a solid solution, rather than the excess gibbs free energy. This commit adds two attributes to solidsolution, activities and activity_coefficients, which output the overall activity and the proportion of this value attributable to nonideality.

sannecottaar commented:
Looks good to me, as far as I understand this. It is spitting out a runtime warning when temperature is 0 for the benchmark_solidsolution.py, so the output should be fixed to pass that test.

bobmyhill commented:
Fixed!

bobmyhill commented:
Ok, should be good to go...

tjhei commented on burnman/solidsolution.py:
```diff
@@ -122,9 +121,30 @@ def set_state(self, pressure, temperature):
+    @material_property
+    def activity_coefficients(self):
+        """
+        Returns a list of endmember activity coefficients (the non-ideal part of the activities
```
closing parens?

tjhei commented on burnman/solidsolution.py:
```diff
@@ -122,9 +121,30 @@ def set_state(self, pressure, temperature):
+    @material_property
+    def activities(self):
+        """
+        Returns a list of endmember activities
```
units?

tjhei commented:
it would also be great to have a test that somehow covers numeric values of these functions.

bobmyhill commented:
Ok, I've changed the implementation to be a bit more efficient (there's no np.exp(np.log()) to calculate activities now). I've also added some tests and added internal and helmholtz energies to the solid solution calculations.

tjhei commented:
I still have no idea what these activities are, but I am not a mineral physicist. :-)

Looks good to me.

bobmyhill commented:
I suspected that this was the reason that it wasn't merged yet :) There's a good introduction here:
http://serc.carleton.edu/research_education/equilibria/activitymodels.html

Don't worry, I'll add some details to the documentation in a separate PR. Activities are very useful.

bobmyhill commented:
@sannecottaar, are you also happy?

sannecottaar commented:
Yes. I'm just going to go ahead and merge this.
